WordPress/Classes/Networking/Pinghub.swift

```diff
@@ -0,0 +1,191 @@
+import Foundation
+import Starscream
+
+
+// MARK: Client
+
+
+/// The delegate of a PinghubClient must adopt the PinghubClientDelegate
+/// protocol. The client will inform the delegate of any relevant events.
+///
+public protocol PinghubClientDelegate {
+    /// The client connected successfully.
+    ///
+    func pingubConnected(client: PinghubClient)
+
+    /// The client disconnected. This might be intentional or due to an error.
+    /// The optional error argument will contain the error if there is one.
+    ///
+    func pinghubDisconnected(client: PinghubClient, error: Error?)
+
+    /// The client received an action.
+    ///
+    func pinghubActionReceived(client: PinghubClient, action: PinghubClient.Action)
+
+    /// The client received some data that it didn't look like a known action.
+    ///
+    func pinghubUnexpectedDataReceived(client: PinghubClient, message: String)
```

Comment:
👍 I just converted it enough that it built, but this makes sense. I might eventually ditch the delegate in favor of callbacks, but I'll take the suggestions for now